movie-api/src/main/java/com/movie/movieapi/interfaces/movie/ScheduleController.java

```diff
@@ -25,4 +24,28 @@ public Response<List<ScheduleDto.Response>> getSchedules(@RequestParam("theaterI
         return Response.success(responses);
     }
 
+    @PostMapping(consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Response<List<ScheduleDto.Response>> getSchedules(@RequestBody ScheduleDto.Search searchDto) {
+        List<ScheduleDto.Response> responses = scheduleService.getSchedules(searchDto.search()).stream()
+                .map(ScheduleDto.Response::from)
+                .toList();
+        return Response.success(responses);
+    }
+
+    @PostMapping(value = "/cached", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Response<List<ScheduleDto.Response>> getSchedulesAsCached(@RequestBody ScheduleDto.Search searchDto) {
+        List<ScheduleDto.Response> responses = scheduleService.getSchedulesAsCached(searchDto.search()).stream()
+                .map(ScheduleDto.Response::from)
+                .toList();
+        return Response.success(responses);
+    }
+
+    @PostMapping(value = "/redis", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Response<List<ScheduleDto.Response>> getSchedulesByRedis(@RequestBody ScheduleDto.Search searchDto) {
+        List<ScheduleDto.Response> responses = scheduleService.getSchedulesAsCached(searchDto.search()).stream()
```

Comment:
```suggestion
        List<ScheduleDto.Response> responses = scheduleService.getSchedulesAsRedis(searchDto.search()).stream()
```

redis 캐시를 호출해야 하는거 아닌가욤?! 실수하셨나 해서 여쭤봅니다

Reply:
실수한거 같습니다.. 아래와 같은 서비스 코드를 보고 있어야 합니다...

```java
public List<ScheduleInfo.Get> getSchedulesByRedis(ScheduleCommand.Search search) {
      List<ScheduleInfo.Get> cachedSchedules = scheduleRedisRepository.find(search);
      List<ScheduleInfo.Get> schedules = new ArrayList<>();
      if (cachedSchedules.isEmpty()) {
          schedules = scheduleRepository.getSchedules(search);
          scheduleRedisRepository.save(search, schedules);
      }
      return schedules;
  }
```